packages/ws-server/src/webSocketServer.ts

```diff
@@ -63,10 +68,53 @@ function getMaxConnectionTTL() {
     return parseInt(process.env.WS_MAX_CONNECTION_TTL || '300000');
 }
 
-app.ws.use((ctx) => {
+// function to get maximum allowed connections from a single IP
+function getLimitPerIp() {
+    return parseInt(process.env.CONNECTION_LIMIT_PER_IP || '10');
+}
+
+async function handleConnectionClose(ctx) {
+    if (ctx.websocket.ipCounted) {
+        const {ip} = ctx.request;
+        clientIps[ip]--;
+        if (clientIps[ip] === 0) delete clientIps[ip];
+    }
+    connectedClients--;
```

Comment:
Q: should the decrement be moved to after the actual unsubscribe is done?
What happens if unsubscribe has a failure, does it still unsubscribe but report the failure or could there be a case where the subscription or connection exists but we decremented?

Reply:
Good point, `unsubscribe` should be executed before all decrements.